vmm/src/config.rs

```diff
@@ -2424,17 +2440,23 @@ mod tests {
             }
         );
 
+        let fd1 =
```

Comment:
Clever hack. It might just be better to open `/dev/null`. I did wonder if this change could be to blame but I couldn't understand how it was affecting the e.g. CPU unit tests. It looks like we we closing other FDs that were being used in the same unit test process and thus causing the erratic behaviour.

Reply:
My attempt to switch to use `/dev/null` failed - it was still showing the "contagious" unit test disruption.

Reply:
> I did wonder if this change could be to blame but I couldn't understand how it was affecting the e.g. CPU unit tests. It looks like we we closing other FDs that were being used in the same unit test process and thus causing the erratic behaviour.

I think so too. My bisected the tree and confirmed that the random failures were caused by changes from #https://github.com/cloud-hypervisor/cloud-hypervisor/pull/5199. I also noticed the failure would happen from a range of tests (not only about cpu unit tests) and they happened randomly.

> My attempt to switch to use /dev/null failed - it was still showing the "contagious" unit test disruption.

Looks like your experiment with `/dev/null` is working? If yes, this is a cleaner way to do it. :)